File: SretneSapice/SretneSapice.Services/Database/Comment.cs
using System;
using System.Collections.Generic;

namespace SretneSapice.Services.Database;

public partial class Comment
{
    public int CommentId { get; set; }

    public int? PostId { get; set; }

    public int? UserId { get; set; }

    public string CommentContent { get; set; } = null!;

    public DateTime? Timestamp { get; set; }

    public int? LikesCount { get; set; }

    public virtual ICollection<CommentLike> CommentLikes { get; } = new List<CommentLike>();

    public virtual ForumPost? Post { get; set; }

    public virtual User? User { get; set; }
}

File: SretneSapice/UI/sretnesapice_mobile/lib/models/comment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sretnesapice_mobile/models/comment_like.dart';
import 'package:sretnesapice_mobile/models/user.dart';

part 'comment.g.dart';

@JsonSerializable()
class Comment {
  int? commentId;
  int? postId;
  int? userId;
  String? commentContent;
  String? timestamp;
  int? likesCount;
  User? user;
  List<CommentLike> commentLikes;

  Comment(this.commentId, this.postId, this.userId, this.commentContent,
      this.timestamp, this.likesCount, this.user, this.commentLikes);

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
}
